ui/src/containers/Visualisations/components/index.js

```diff
@@ -0,0 +1,19 @@
+import AddQueryButton from './AddQueryButton';
```

Comment:
I probably wouldn't use index files if possible, I'd rather just import the components directly to reduce a bit of code maintenance. Not sure what you think?

Reply:
I think importing from index file is simpler for developers.

The current CustomBarChart/Editor.js is like below

```
import {
  AddQueryButton,
  BarChartGroupingLimitForm,
  DescriptionForm,
  PreviewPeriodPicker,
  SourceViewForm,
  StackedSwitch,
  TimezoneForm,
} from '../components';
```

Other chart files will import many components from '../components' directory.

If we delete the index file and import from each component files, CustomBarChart/Editor.js will be

```
import AddQueryButton from '../components/AddQueryButton';
import BarChartGroupingLimitForm from '../components/BarChartGroupingLimitForm';
import DescriptionForm from '../components/DescriptionForm';
import PreviewPeriodPicker from '../components/PreviewPeriodPicker';
import SourceViewForm from '../components/SourceViewForm';
import StackedSwitch from '../components/StackedSwitch';
import TimezoneForm from '../components/TimezoneForm';
```

Reply:
Yeah definitely shorter and simpler to read. Just considering the experience for adding, removing, and renaming components.

When you add a component you might not be aware of the need to change the index file so we lose some consistency (although we have code reviews to hopefully catch that).

When you remove the component, again you need to know to delete the component file and export in the index, you also have to potentially check for references via the index file and the component file just in case someone forgets to import it from the index.

When you rename the component, you have to rename it in the component file, then find all the references to it via the index file and the component file. You also have to rename it in the index file too.

I think it's okay, not too much extra maintenance effort, just something we'll have to be weary of in code reviews to keep the consistency.

Reply:
OK. I follow the style.